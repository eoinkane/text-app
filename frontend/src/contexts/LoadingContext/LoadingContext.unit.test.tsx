import React from "react";
import { mount } from "enzyme";
import { useLoading, LoadingContextProvider, types, reducer } from "./index";

const TestHelper = () => {
  const { loading, loadingActions } = useLoading("TestHelper");
  return (
    <div>
      <p>{loading ? "true" : "false"}</p>
      <button onClick={loadingActions.ON}>ON</button>
      <button onClick={loadingActions.OFF}>OFF</button>
    </div>
  );
};

describe("Loading Context", () => {
  it("Renders children", () => {
    const loadingContext = mount(
      <LoadingContextProvider>
        <TestHelper />
      </LoadingContextProvider>
    );
    expect(loadingContext).toMatchSnapshot();

    expect(
      loadingContext.find(LoadingContextProvider).children().first().type()
    ).toBe(TestHelper);
  });

  it("Provides loading to the component", () => {
    const loadingContext = mount(
      <LoadingContextProvider>
        <TestHelper />
      </LoadingContextProvider>
    );

    expect(loadingContext.find("p").text()).toBe("false");
  });

  it("Dispatch ON changes loading to true", () => {
    const loadingContext = mount(
      <LoadingContextProvider>
        <TestHelper />
      </LoadingContextProvider>
    );

    loadingContext.find("button").first().simulate("click");

    expect(loadingContext.find("p").text()).toBe("true");
  });

  it("Dispatch OFF changes loading to false", () => {
    const loadingContext = mount(
      <LoadingContextProvider>
        <TestHelper />
      </LoadingContextProvider>
    );

    loadingContext.find("button").first().simulate("click");
    loadingContext.find("button").last().simulate("click");

    expect(loadingContext.find("p").text()).toBe("false");
  });
});

describe("Loading Reducer", () => {
  it("ON action", () => {
    const result = reducer(false, { type: types.ON });

    expect(result).toMatchSnapshot();

    expect(result).toBe(true);
  });

  it("OFF action", () => {
    const result = reducer(true, { type: types.OFF });

    expect(result).toMatchSnapshot();

    expect(result).toBe(false);
  });
});
