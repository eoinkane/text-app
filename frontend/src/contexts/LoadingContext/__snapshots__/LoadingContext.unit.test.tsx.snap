// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Context Renders children 1`] = `
<LoadingContextProvider>
  <TestHelper>
    <div>
      <p>
        false
      </p>
      <button
        onClick={[Function]}
      >
        ON
      </button>
      <button
        onClick={[Function]}
      >
        OFF
      </button>
    </div>
  </TestHelper>
</LoadingContextProvider>
`;

exports[`Loading Reducer OFF action 1`] = `false`;

exports[`Loading Reducer ON action 1`] = `true`;
